Site Settings: hook and add the Jetpack Disconnect Site

AnnaMag opened the pull request:
This is a landing PR for the new Jetpack Disconnect Flow. It introduces a basic layout of the first page of the flow and makes it visible from the /settings/disconnect-site page (in dev). 

Included:
- Introduce the `Disconnect Site`, hook it up and add route definitions
- add style and included it for loading
- add `Skip Survey` component, which will redirect to the end point of the flow (`Confirm Disconnection` https://github.com/Automattic/wp-calypso/pull/18039). For the time being, it redirects to `/settings/manage-connection`.
- add dev feature flag 
- add Placeholder for the Disconnect Site to be shown with an empty Redux state 
Placeholder will be corrected in a separate PR https://github.com/Automattic/wp-calypso/pull/18127 as an independent component or appropriate styling.

The PR uses `redirectNonJetpack` HoC  https://github.com/Automattic/wp-calypso/pull/17555 for access control.

**To test**:
- Checkout this branch, or use calypso.live
 https://calypso.live/?branch=add/site-settings-disconnect-site-layout

For Jetpack sites (incl. the dev mode):
 - verify that the start of the flow is visible at ```/settings/disconnect-site/:site```, where ```:site``` is the Jetpack site.

For non-Jetpack and Atomic sites, 
 - verify immediate redirection to `/settings/general/:site`

With empty Redux state we should see a placeholder prior to rendering.

**Visuals:**
After removing the sidebar:
![screen shot 2017-09-23 at 22 28 36](https://user-images.githubusercontent.com/13561163/30777449-b5d262ce-a0b2-11e7-99c7-994531670b2c.png)

AnnaMag commented:
It is a trimmed and clean version of https://github.com/Automattic/wp-calypso/pull/17155, which went out of control (it was early PR and IMHO better leave it as a reference).

sirreal commented on client/my-sites/site-settings/disconnect-site/skip-survey.jsx:
```diff
@@ -0,0 +1,42 @@
+
+		//placeholder for redirection to the Confirm Disconnection page
+		if ( siteSlug ) {
+			page( '/settings/general/' + siteSlug );
```
I know this will be updated, but `/settings/manage-connection` is probably less confusing for now.

AnnaMag commented:
@sirreal thanks so much for reviewing the code!

I addressed the comments 👍  I took the disconnect flow specific Placeholder out of this PR altogether and use a 'dummy' one for the time being. I'd prefer to handle it separately in the follow-up  review to ease the review process.

sirreal commented on client/my-sites/site-settings/index.js:
```diff
@@ -2,7 +2,6 @@
  * External dependencies
  */
 import page from 'page';
-
```
There was nothing wrong with this empty line 😉

AnnaMag commented:
@sirreal thanks a lot! This patch is complete and would be great to see it merged 👍

sirreal commented:
Good catch on that case with the sidebar! 🚀

tyxla commented:
Thanks @AnnaMag and @sirreal!

sirreal commented:
@AnnaMag This branch can't be deleted because it's associated with another open PR. I haven't found it, but that seems pretty strange. Do you know what's going on?

AnnaMag commented:
@sirreal I built follow-up PRs on top of it. It is weird but I have more open PRs than those that are visible when filtering by the person assigned 🤔 It is probably the reason why you cannot see it. Let me rebase the others against master and free this one. Will let you know once ready (on it now). Thanks!
